### PR TITLE
feat(tasks): add expand flag and executionIds param to /pipelines end…

### PR DESCRIPTION
**Pull request by anotherchrisberry**
…point

A few changes here to support work on the Teams page views. Two should not be controversial, the third may be.

**1. Security**
Perform a post-filter operation to make sure the user has the rights to read the executions

**2. `expand` parameter**
Defaults to `true` for backwards compatibility; does the same thing as the other methods in the controller with the same parameter

**3. `executionIds` parameter**
This one is a little more questionable, as I'm not sure it's all that intuitive to understand the endpoint without reading the documentation - you need to supply either this parameter or `pipelineConfigIds`, but if you use this one, some other parameters are going to be ignored.

### CODE REVIEW

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy [hunk 1]
```diff
@@ -178,27 +178,65 @@ class TaskController {
...
+      List<Execution> executions = rx.Observable.from(ids.collect {
+        executionRepository.retrieve(PIPELINE, it)
+      }).subscribeOn(Schedulers.io()).toList().toBlocking().single()
 
+      if (!expand) {
+        unexpandPipelineExecutions(executions)
+      }
+
+      return executions
+    } else {
```
Comment (ajordens) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy]: doesn't seem like you need the actual `else`?

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy [hunk 2]
```diff
@@ -178,27 +178,65 @@ class TaskController {
...
     statuses = statuses ?: ExecutionStatus.values()*.toString().join(",")
     limit = limit ?: 1
-    def executionCriteria = new ExecutionRepository.ExecutionCriteria(
+    ExecutionRepository.ExecutionCriteria executionCriteria = new ExecutionRepository.ExecutionCriteria(
       limit: limit,
       statuses: (statuses.split(",") as Collection)
     )
 
-    def ids = pipelineConfigIds.split(',')
+    if (!pipelineConfigIds && !executionIds) {
```
Comment (ajordens) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy]: what happens if both `pipelineConfigIds` _and_ `executionIds` are supplied ... seems like we'll only take the `executionIds`?

Comment (anotherchrisberry) [reply orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy]: that's correct - I documented that behavior in the javadoc above

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy (hunk 1 above)
Comment (anotherchrisberry) [reply orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy]: True, I'll remove.

**Pull request merged**